src/helpers/script_helpers.sh

```diff
@@ -109,9 +109,15 @@ function run {
     undo_command
   else
     remove_step_files
+
+    # Running inside a subfolder can cause issues if that folder doesn't exist
+    # on the main branch. Hence we cd into the root directory of the Git repo here.
+    # See https://github.com/Originate/git-town/issues/371 for more details.
+    cd_git_root
     preconditions "$@"
     steps > "$STEPS_FILE"
     run_steps "$STEPS_FILE" undoable
+    cd_start_dir
```

Comment:
Great point. I have refactored the solution to make cd'ing into the root directory a real, undoable, logged step. This not only displays to the user that Git Town was cd'ing out of the current directory, it also properly goes back to the directory in all situations. Specs for all cases added.